### DISCUSSION TITLE
Production - Cache instruction not applied

### DISCUSSION THREAD

**Discussion by NicolasMarafetti** [vercel/next.js#38947] [Help] [answered]
**What version of Next.js are you using?**

12.2.2

**What version of Node.js are you using?**

v16.15.0 (local) v16.16.0 (on prod)

**What browser are you using ?**

Google Chrome

**What operating system are you using?**

Windows

**How are you deploying your application**

next build
pm2 start npm

**Server Configuration**

My server is a VPS on Debian16 and running with Apache2

**Describe the Bug**

Building an e-commerce website. 
On production, when I go to the cart page, I always have the same cart, even after adding / removing products.
This is because I have the default Cache-Control for the _next/data/development/cart.json file. This is the file which return my getServerSideProps results.
If I press CTRL + R, the cart page is now at correct state. But if I leave the page and come back, I always have the first version stored in cache.

Here is my getServerSideProps function for the cart page :

```
export const getServerSideProps = withIronSessionSsr(async (context) => {
  // Get the cart
  let cart: CartInterfaceForCartPage | null = null;
  if (context.req.session.user) {
    cart = await getCartFromDatabaseForCartPage(context.req.session.user.id);
  }

  context.res.setHeader(
    'Cache-Control',
    'no-cache, no-store, max-age=0, must-revalidate'
  );

  return {
    props: {
      cart,
    },
  };
}, ironOptions);
```

withIronSessionSsr is a module to have session on the server side.

**On Local**
With npm run dev or build & start, the cart page is working nicely and I have for the cart.json file :
Status Code: 200 OK
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
like I asked in my getServerSideProps function.

**On Production**

With npm run dev or build & start the cart.json ressource have these headers  :
Status Code: 200 OK (from disk cache)
Cache-Control: max-age=31536000, public
which is the default Next cache control (I think).

**Comment by wwwdepot** [comment, marked as answer]
Make sure your proxy is not overriding the cache-control headers.
